pytrip/spc.py

```diff
@@ -0,0 +1,547 @@
+                # At present nC=0.
+                tag = Tag(fd)
+                tag.code = 15
+                tag.size = 8
+                tag.endian = self.endian
+                tag.set_tag()
+                fd.write(np.asarray([specie.nc], dtype="{:s}u{:d}".format(tag._ste, tag.size)))
+
+                # no 16, 8-byte unsigned integer;
+                # 8-byte unsigned integer; number of energy bins for this species
+                tag = Tag(fd)
+                tag.code = 16
+                tag.size = 8
+                tag.endian = self.endian
+                tag.set_tag()
+                fd.write(np.asarray([specie.ne], dtype="{:s}u{:d}".format(tag._ste, tag.size)))
+
+                # If a species data block is flagged as EREFCOPY, its energy bins are not stored,
+                # but rather a reference index <lSRef> (0..<nS>-1) to a species with identical energy binning.
+                # This helps to reduce space requirements, since fragment spectra may share the same energy bins.
+                # If the energy bins for the species under consideration are not a reference copy,
+                # <nE>+1 double precision bin border values are stored.
+                compatible_ebins_index = [i for i in ebins if np.array_equal(specie.ebindata, ebins[i])]
+                if not compatible_ebins_index:
+                    ebins[l] = specie.ebindata
+
+                    # no 17, double; energy bin values
+                    tag = Tag(fd)
+                    tag.code = 17
+                    tag.size = 8 * (specie.ne + 1)
+                    tag.endian = self.endian
+                    tag.set_tag()
+                    fd.write(np.asarray(specie.ebindata, dtype="{:s}f8".format(tag._ste)))
+                else:
+                    # no 18,  8-byte unsigned integer
+                    tag = Tag(fd)
+                    tag.code = 18
+                    tag.size = 8
+                    tag.endian = self.endian
+                    tag.set_tag()
+                    fd.write(np.asarray([compatible_ebins_index[0]], dtype="{:s}u{:d}".format(tag._ste, tag.size)))
+
+                # no 19,  double; spectrum bin values
+                tag = Tag(fd)
+                tag.code = 19
+                tag.size = 8 * specie.ne
+                tag.endian = self.endian
+                tag.set_tag()
+                fd.write(np.asarray(specie.histdata, dtype="{:s}f8".format(tag._ste)))
+
+                # no 20,  double; running cumulated spectrum bin values
+                tag = Tag(fd)
+                tag.code = 20
+                tag.size = 8 * (specie.ne + 1)
+                tag.endian = self.endian
+                tag.set_tag()
+                fd.write(np.asarray(specie.rcumdata, dtype="{:s}f8".format(tag._ste)))
+
+        fd.close()
+
+    def read_data(self):
+        fd = open(self.filename, "rb")
+        t = Tag(fd)
+        db = []
+
+        while t.code < 9:
```

Comment:
how about using all caps named var for this to make it more readable?
t.code < SPC_TAG_FOOBAR:

Reply:
using a dict of tags

Reply:
See proposed solution here 19aa2c1